solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -513,6 +519,54 @@ pub mod solana_ibc {
         let chain = &mut ctx.accounts.chain;
         chain.update_chain_config(config_payload)
     }
+
+    /// Method which updates the connection delay of a particular connection
+    ///
+    /// Fails if the connection doesnt exist.
+    /// Can only be called by fee collector.
+    pub fn update_connection_delay_period(
+        ctx: Context<UpdateConnectionDelay>,
+        connection_id_idx: u64,
+        delay_period_in_ns: u64,
+    ) -> Result<()> {
+        let storage = &mut ctx.accounts.storage;
+
+        let connection_id = ConnectionId::new(connection_id_idx);
+
+        // Panic if connection_id doenst exist
+        if storage.connections.len() >= connection_id_idx as usize {
```

Comment:
I’m not a fan of `as` casting, though I suppose there’s no obvious way
to replace it.  We could change `connection_id_idx` to `u16` and then
use `usize::from(connection_id_idx)` or do
`usize::try_from(connection_id_idx).unwrap()`.  Not sure if either of
those is actually better.

Reply:
if we use u16, we would need to convert it again to get `connection_id`

Reply:
Yes, but you can do it with just `_.into()` or `u64::from(_)`.

Reply:
but what is the benefit of getting rid of `as` casting. does it affect the performance?

Reply:
oh i see. so it makes sense to send it as u16 and use `from` and `into` for conversions then.